core/src/rigidbody.cpp

```diff
@@ -0,0 +1,289 @@
+#include <knoting/rigidbody.h>
+
+namespace knot {
+namespace components {
+RigidBody::RigidBody() : m_physics(NULL), m_dynamic(NULL), m_static(NULL), m_scene(NULL), m_material(NULL) {}
+
+RigidBody::~RigidBody() {
+    if (m_dynamic) {
+        m_dynamic->release();
+        m_dynamic = NULL;
+    } else {
+        m_static->release();
+        m_static = NULL;
+    }
+    m_material->release();
+    m_material = NULL;
+    m_scene = NULL;
+    m_physics = NULL;
+}
+
+void RigidBody::on_awake() {}
+void RigidBody::on_destroy() {
+    if (m_dynamic) {
+        m_scene->removeActor(*m_dynamic);
+    } else if (m_static) {
+        m_scene->removeActor(*m_static);
+    }
+}
+
+vec3 RigidBody::get_position() const {
+    vec3 v;
+    if (m_dynamic) {
+        v = vec3(m_dynamic->getGlobalPose().p.x, m_dynamic->getGlobalPose().p.y, m_dynamic->getGlobalPose().p.z);
+
+    } else {
+        v = vec3(m_static->getGlobalPose().p.x, m_static->getGlobalPose().p.y, m_static->getGlobalPose().p.z);
+    }
+    return v;
```

Comment:
Without compiler optimizations, you are allocating the `vec3 v` twice.

```suggestion
    if (m_dynamic) {
        return vec3(m_dynamic->getGlobalPose().p.x, m_dynamic->getGlobalPose().p.y, m_dynamic->getGlobalPose().p.z);

    } else {
        return vec3(m_static->getGlobalPose().p.x, m_static->getGlobalPose().p.y, m_static->getGlobalPose().p.z);
    }
    return vec3(0.0f);
```